### PR TITLE
[BUGFIX] Turtle-Service Reset Method Crashes Wallet

### LINKED ISSUE #547: [BUG] reset through service RPC with viewSecretKey argument result in wallet being corrupted
### System Information

**Operating System**: Ubuntu
**OS Version**: 16.04.5 LTS
**CPU Make, Model, & Speed**: Intel i7-6700k 4.0Ghz
**Amount of RAM**: 8GB
**Hard Drive Type**: Mechanical HDD
**Hard Drive Size**: 1TB
**Volume Type**: lvm2 pv

### TurtleCoin Build Information

**Version:** 0.8.4.1276
**Binary Source:** Self Compiled
```
mkdir build && cd build
cmake .. 
make TurtleCoind -j4
make service -j4
make zedwallet -j4
```

**TurtleCoin CMake output at the default verbosity**:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found Git: /usr/bin/git
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   thread
--   date_time
--   chrono
--   regex
--   serialization
--   program_options
--   atomic
-- Boost Found: /usr/include
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- Enabling RTTI in Debug builds only (default)
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- JNI library is disabled
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rashed/github/test1/build/rocksdb
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rashed/github/test1/build
```

### Issue Report

**Binary Reported:** turtle-service

#### Expected Behavior

If the viewSecretKey argument is provided, the reset() method substitutes the existing wallet with a new one with the specified key.

#### Current Behavior

Leaves the wallet in a corrupted state.

##### Screen Shots

selection_001

#### Steps to Reproduce

1. open a wallet with turtle-service
2. check through getStatus RPC call to see if it is syncing.. it will
3. Now send reset RPC call to the turtle-service from another terminal with viewSecretKey of the same wallet.
4. check again through getStatus RPC call to see if it is syncing.. it won't - will be stuck on 1 forever.
5. try opening the wallet with zedwallet.. you can't- it gives the errors shown in above screenshot

### PR DESCRIPTION
**Pull request by brandonlehmann**
Fixed #547 

If you want to swap view keys (as in swap everything in a container to a new set of keys) create a new container.

### CODE REVIEW

File: src/WalletService/PaymentServiceJsonRpcServer.cpp [hunk 1]
```diff
@@ -123,11 +123,7 @@ std::error_code PaymentServiceJsonRpcServer::handleExport(const Export::Request&
 }
 
 std::error_code PaymentServiceJsonRpcServer::handleReset(const Reset::Request& request, Reset::Response& response) {
-  if (request.viewSecretKey.empty()) {
-    return service.resetWallet(request.scanHeight);
-  } else {
-    return service.replaceWithNewWallet(request.viewSecretKey, request.scanHeight, request.newAddress);
```
Comment (zpalmtree) [inline src/WalletService/PaymentServiceJsonRpcServer.cpp]: Can we remove `replaceWithNewWallet` now?

Comment (brandonlehmann) [reply src/WalletService/PaymentServiceJsonRpcServer.cpp]: Probably. Wait one. Fixing #548 now lol.

Comment (brandonlehmann) [reply src/WalletService/PaymentServiceJsonRpcServer.cpp]: Removed

**Comment by brandonlehmann** [comment]
We good?

**Comment by zpalmtree** [comment]
Need to update docs :disappointed:

**Pull request merged**